### ISSUE TITLE
Improved TypeScript support

### ISSUE THREAD

**Issue by fabiospampinato** [petyosi/react-virtuoso#261]
I'm having troubles typing well this component:

```ts
const Scroller = useCallback ( forwardRef ( ( { style, children }, ref ) => {
  return (
    <div style={style} ref={ref}>
      {children}
    </div>
  );
}), [] );
```

There should be TypeScript types importable from the 'react-virtuoso' package for each of these components Virtuoso accepts, and types for each props object they accept, without them overriding a component and typing it well is way more difficult than it should be.

**Comment by petyosi** [comment]
Agreed, typings are an improvement.

**Comment by petyosi** [comment]
Addressed in the 6b3413e7f437db79b489bc873a1dd45e9c100d75, released in v1.2.7.
